_notebooks/2020-07-28-lineage.ipynb

```diff
@@ -0,0 +1,323 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# To use the latest publish `kubeflow-metadata` library, you can run:\n",
+    "!pip install kubeflow-metadata --user\n",
+    "# Install other packages:\n",
+    "!pip install pandas --user\n",
+    "# Then restart the Notebook kernel."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas\n",
+    "from kubeflow.metadata import metadata\n",
+    "from datetime import datetime\n",
+    "from uuid import uuid4"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "parameters"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "METADATA_STORE_HOST = \"metadata-grpc-service.kubeflow\" # default DNS of Kubeflow Metadata gRPC serivce.\n",
+    "METADATA_STORE_PORT = 8080"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create a workspace and a run in a workspace\n",
+    "* [Workspace](https://github.com/kubeflow/metadata/blob/25b44da29213968a2c438d24aad3656cc86d0499/sdk/python/kubeflow/metadata/metadata.py#L92) groups a set of runs of pipelines, notebooks and their related artifacts and executions\n",
+    "* [Store](https://github.com/kubeflow/metadata/blob/25b44da29213968a2c438d24aad3656cc86d0499/sdk/python/kubeflow/metadata/metadata.py#L59) that connects to the Metadata gRPC service\n",
+    "* [Run](https://github.com/kubeflow/metadata/blob/25b44da29213968a2c438d24aad3656cc86d0499/sdk/python/kubeflow/metadata/metadata.py#L227) captures a run of pipeline or notebooks in a workspace and group execution"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ws1 = metadata.Workspace(\n",
+    "    # Connect to metadata service in namespace kubeflow in k8s cluster.\n",
+    "    store=metadata.Store(grpc_host=METADATA_STORE_HOST, grpc_port=METADATA_STORE_PORT),\n",
+    "    name=\"xgboost-synthetic\",\n",
+    "    description=\"workspace for xgboost-synthetic artifacts and executions\",\n",
+    "    labels={\"n1\": \"v1\"})"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "r = metadata.Run(\n",
+    "    workspace=ws1,\n",
+    "    name=\"xgboost-synthetic-faring-run\" + datetime.utcnow().isoformat(\"T\") ,\n",
+    "    description=\"a notebook run\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create an execution in a run\n",
+    "* [Execution](https://github.com/kubeflow/metadata/blob/25b44da29213968a2c438d24aad3656cc86d0499/sdk/python/kubeflow/metadata/metadata.py#L251) captures a run of pipeline or notebooks in a workspace and group executions. Execution also serves as object for logging artifacts as its input or output"
```

Comment:
I would explain more clearly what is the relationship between a Run and an Execution here. Maybe you want to say that an Execution is a specific "Instance" of a run? And that you can "bind" specific input/output artifacts to this instance?